### PR TITLE
DO signature generator: Add support for abstract/interface IId types

### PR DESCRIPTION
**Pull request by paolobazzi**
Added new type "IDI" representing abstract IId classes and custom IId interfaces.

322156

### CODE REVIEW

File: org.eclipse.scout.rt.dataobject.test/src/test/java/org/eclipse/scout/rt/dataobject/testing/signature/DataObjectSignatureGeneratorTest.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+/**
+ * Test cases for {@link DataObjectSignatureGenerator} and {@link DataObjectSignatureComparator}.
+ */
+public class DataObjectSignatureGeneratorTest {
+
+  @Test
+  public void testCreateSignature() {
+    DataObjectSignatureDo signature = BEANS.get(DataObjectSignatureGenerator.class).createSignature(CollectionUtility.hashSet(getPackageNamePrefix()), c -> true, (c, s) -> true);
+    System.out.println(signature);
```
Comment (stephan-merkli) [inline org.eclipse.scout.rt.dataobject.test/src/test/java/org/eclipse/scout/rt/dataobject/testing/signature/DataObjectSignatureGeneratorTest.java]: Remove System.out

File: org.eclipse.scout.rt.dataobject.test/src/test/java/org/eclipse/scout/rt/dataobject/testing/signature/fixture/SignatureFixtureDo.java [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+import org.eclipse.scout.rt.dataobject.DoValue;
+import org.eclipse.scout.rt.dataobject.TypeName;
+import org.eclipse.scout.rt.dataobject.TypeVersion;
+import org.eclipse.scout.rt.dataobject.fixture.DataObjectFixtureTypeVersions;
+import org.eclipse.scout.rt.dataobject.fixture.FixtureEnum;
+import org.eclipse.scout.rt.dataobject.fixture.FixtureStringId;
+import org.eclipse.scout.rt.dataobject.id.AbstractStringId;
+
+@TypeName("scout.SignatureFixture")
+@TypeVersion(DataObjectFixtureTypeVersions.DataObjectFixture_1_0_0.class)
```
Comment (stephan-merkli) [inline org.eclipse.scout.rt.dataobject.test/src/test/java/org/eclipse/scout/rt/dataobject/testing/signature/fixture/SignatureFixtureDo.java]: static import

File: org.eclipse.scout.rt.dataobject.test/src/test/java/org/eclipse/scout/rt/dataobject/testing/signature/fixture/SignatureSubFixtureDo.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+import javax.annotation.Generated;
+
+import org.eclipse.scout.rt.dataobject.DoEntity;
+import org.eclipse.scout.rt.dataobject.DoValue;
+import org.eclipse.scout.rt.dataobject.TypeName;
+import org.eclipse.scout.rt.dataobject.TypeVersion;
+import org.eclipse.scout.rt.dataobject.fixture.DataObjectFixtureTypeVersions;
+
+@TypeName("scout.SignatureSubFixture")
+@TypeVersion(DataObjectFixtureTypeVersions.DataObjectFixture_1_0_0.class)
```
Comment (stephan-merkli) [inline org.eclipse.scout.rt.dataobject.test/src/test/java/org/eclipse/scout/rt/dataobject/testing/signature/fixture/SignatureSubFixtureDo.java]: static import
Use version DataObjectFixture_1_0_0_034 just as an additional test (and change in expected)?

File: org.eclipse.scout.rt.dataobject.test/src/test/java/org/eclipse/scout/rt/dataobject/testing/signature/fixture/SignatureSubFixtureDo.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+import javax.annotation.Generated;
+
+import org.eclipse.scout.rt.dataobject.DoEntity;
+import org.eclipse.scout.rt.dataobject.DoValue;
+import org.eclipse.scout.rt.dataobject.TypeName;
+import org.eclipse.scout.rt.dataobject.TypeVersion;
+import org.eclipse.scout.rt.dataobject.fixture.DataObjectFixtureTypeVersions;
+
+@TypeName("scout.SignatureSubFixture")
```
Comment (stephan-merkli) [inline org.eclipse.scout.rt.dataobject.test/src/test/java/org/eclipse/scout/rt/dataobject/testing/signature/fixture/SignatureSubFixtureDo.java]: use correct namespace here (matching type version namespace): dataObjectFixture

File: org.eclipse.scout.rt.dataobject.test/src/test/java/org/eclipse/scout/rt/dataobject/testing/signature/fixture/SignatureFixtureDo.java [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+import org.eclipse.scout.rt.dataobject.DoList;
+import org.eclipse.scout.rt.dataobject.DoValue;
+import org.eclipse.scout.rt.dataobject.TypeName;
+import org.eclipse.scout.rt.dataobject.TypeVersion;
+import org.eclipse.scout.rt.dataobject.fixture.DataObjectFixtureTypeVersions;
+import org.eclipse.scout.rt.dataobject.fixture.FixtureEnum;
+import org.eclipse.scout.rt.dataobject.fixture.FixtureStringId;
+import org.eclipse.scout.rt.dataobject.id.AbstractStringId;
+
+@TypeName("scout.SignatureFixture")
```
Comment (stephan-merkli) [inline org.eclipse.scout.rt.dataobject.test/src/test/java/org/eclipse/scout/rt/dataobject/testing/signature/fixture/SignatureFixtureDo.java]: use correct namespace here (matching type version namespace): dataObjectFixture

File: org.eclipse.scout.rt.dataobject.test/src/test/java/org/eclipse/scout/rt/dataobject/testing/signature/DataObjectSignatureGeneratorTest.java (hunk 1 above)
Comment (paolobazzi) [reply org.eclipse.scout.rt.dataobject.test/src/test/java/org/eclipse/scout/rt/dataobject/testing/signature/DataObjectSignatureGeneratorTest.java]: Done

File: org.eclipse.scout.rt.dataobject.test/src/test/java/org/eclipse/scout/rt/dataobject/testing/signature/fixture/SignatureFixtureDo.java (hunk 2 above)
Comment (paolobazzi) [reply org.eclipse.scout.rt.dataobject.test/src/test/java/org/eclipse/scout/rt/dataobject/testing/signature/fixture/SignatureFixtureDo.java]: Done

File: org.eclipse.scout.rt.dataobject.test/src/test/java/org/eclipse/scout/rt/dataobject/testing/signature/fixture/SignatureSubFixtureDo.java (hunk 3 above)
Comment (paolobazzi) [reply org.eclipse.scout.rt.dataobject.test/src/test/java/org/eclipse/scout/rt/dataobject/testing/signature/fixture/SignatureSubFixtureDo.java]: Done

File: org.eclipse.scout.rt.dataobject.test/src/test/java/org/eclipse/scout/rt/dataobject/testing/signature/fixture/SignatureSubFixtureDo.java (hunk 4 above)
Comment (paolobazzi) [reply org.eclipse.scout.rt.dataobject.test/src/test/java/org/eclipse/scout/rt/dataobject/testing/signature/fixture/SignatureSubFixtureDo.java]: Done

File: org.eclipse.scout.rt.dataobject.test/src/test/java/org/eclipse/scout/rt/dataobject/testing/signature/fixture/SignatureFixtureDo.java (hunk 5 above)
Comment (paolobazzi) [reply org.eclipse.scout.rt.dataobject.test/src/test/java/org/eclipse/scout/rt/dataobject/testing/signature/fixture/SignatureFixtureDo.java]: Done

**Pull request merged**